freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/conversion/UserMetadataConverter.java

```diff
@@ -0,0 +1,100 @@
+package com.sequenceiq.freeipa.service.freeipa.user.conversion;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.google.common.base.Strings;
+import com.google.common.io.BaseEncoding;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.auth.altus.CrnParseException;
+import com.sequenceiq.freeipa.client.model.User;
+import com.sequenceiq.freeipa.service.freeipa.user.model.JsonUserMetadata;
+import com.sequenceiq.freeipa.service.freeipa.user.model.UserMetadata;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserMetadataConverter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserMetadataConverter.class);
+
+    private static final ObjectReader OBJECT_READER = new ObjectMapper().readerFor(JsonUserMetadata.class)
+            .without(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+
+    private static final ObjectWriter OBJECT_WRITER = new ObjectMapper().writerFor(JsonUserMetadata.class);
+
+    @SuppressWarnings("checkstyle:NPathComplexity")
+    public Optional<UserMetadata> toUserMetadata(User user) {
```

Comment:
this method is really hard to read with all the return statements. I just tossed some ideas together here (I don't really like my null checks...), but can you try to clarify this a bit (hopefully to the point you won't need to suppress NPathComplexity)?

```
JsonUserMetadata jsonUserMetadataOptional = getJsonUserMetadata(user);
if (jsonUserMetadata != null &&
    hasValidCrn(jsonUserMetadata) {
  UserManagementProto.WorkloadUserSyncActorMetadata protoMetadata = getProtoMeta(jsonUserMetadata);
  if (protoMetadata != null) {
    return Optional.of(new UserMetadata(jsonUserMetadata.getCrn(), protoMetadata.getWorkloadCredentialsVersion()));
}
return Optional.empty();
```

Reply:
I find the "early return" style in a case like this more readable than a series of nested if/else blocks. I did rewrite this method to avoid the checkstyle warning, but felt that made it more difficult to follow the logic, and kept it as it is.

However, since the checkstyle warning indicates that the Cloudbreak style convention is to avoid multiple return statements, I'll rewrite it to conform to that.

Reply:
Done.